src/theory/arrays/theory_arrays_rewriter.cpp

```diff
@@ -76,6 +80,29 @@ Node TheoryArraysRewriter::rewriteViaRule(ProofRewriteRule id, const Node& n)
 {
   switch (id)
   {
+    case ProofRewriteRule::MACRO_ARRAYS_DISTINCT_ARRAYS:
+    {
+      if (n.getKind() == Kind::EQUAL && n[0].isConst() && n[1].isConst()
+          && n[0] != n[1])
+      {
+        Assert(n[0].getType().isArray());
+        return d_nm->mkConst(false);
+      }
+    }
+    break;
+    case ProofRewriteRule::MACRO_ARRAYS_NORMALIZE_CONSTANT:
+    {
+      if (n.getType().isArray() && n[0].isConst() && n[1].isConst()
+          && n[2].isConst())
```

Comment:
Is this a `store` operation? If so, how about just `n.isConst()`? If not, then I am worried about index out-of-bound error

Reply:
`n.isConst` is only true if the STORE is *canonical*.

I will test whether this is a store.